Tasks/DotNetCoreInstallerV1/Task Design.md

```diff
@@ -0,0 +1,110 @@
+- User can choose to install a particular version in isolation of all other version already installed, by specifying a path which is cleaned up after each release/build.
+- User can augment existing set of versions installed with another version by specifying a path where a set of .Net core versions might already be installed, such as: $(Agent.ToolsDirectory)
+- User can also install it at global .Net Core installation path, such as: C:\Program Files\dotnet or /usr/share/dotnet/. This way all agents running on that machine will have access to the newly added .Net core versions.
+
+The way a version is installed at the `installationPath` is now as follows:
+- We extract all folders from dotnetVersions.zip/tgz into the `installationPath`, just like install-dotnet.ps1/sh.
+- The files in root directory of archive are only copied/overriden if
+  - The .Net core version being installed at `installationPath` is later than all other versions already installed at the path.
+
+The advantage of the above installation approach are:
+- Time is saved by not copying unneeded files.
+- Failures which might occur while overriding existing files are avoided.
+
+Also to furhter optimize and save time while installation, we do not install a version if it is already cached at that path. Below described are different approaches to find out if a version is cached,
+
+1. **`Folder based approach`**
+   - `Installation process:` While installing a version, we will only extract the files from downloaded archive and then copy paste the files into the installationPath folder.
+   - `Caching process:` Existance of folder with name = version inside sdk or host/fxr folder will be considered as proof that sdk/runtime version is cached. Alogirthm to find if version is cached or not will be like this:
+     - `installationPath` should contains a folder called `sdk or host/fxr`. If it
+       - `exists:` check if sdk or host/fxr folder contains a folder with name = version to be downloaded.
+         - `exists:` Version is cached, nothing more needs to be done.
+         - `doesn't exist:` Version needs to be installed.
+       - `doesn't exist:` Version needs to be installed
+
+2. **`version.completed file based approach`**
+   - `Installation process:` Once all the files are successfully extracted, copied and pasted into installationPath, we will create a **`version.completed`** file in sdk/runtime folder. The version.completed file will exist at the same level as the version folder and will be considered as a stamp of successfull installation.
+     - If installation is not complete/successfull, we will not create the version.completed file.
+   - `Caching Process:` So, the algorithm will be same as in Approach 1, but with an additional required check for `version.completed` file to be present in sdk or host/fxr folder along with folder with name = version. Based on version.completed's existance, the following will be concluded
```

Comment:
SDk brings the runtime with itself. Will you also create a marker file in host/fxr?

Reply:
or you could find corresponding runtime version and create one marker file for that(could be at top folder itself)

Reply:
doing it for the runtime as well when installing sdk in shared/Microsoft.NETCore.App